onevizion.py

```diff
@@ -1569,6 +1572,281 @@ def sendmail(self):
+        """
+        pass
+
+    def _prepareNotifQueue(self, notifQueue):
+        return notifQueue
+
+    
+class NotifQueueApi:
+    """Wrapper for calling the Onvizion API for Notification Queue. You can get a Notifications Queue, 
+        update the status of a notification queue record, add new attempt 
+
+    Attributes:
+        serviceId: ID of the Notification Service
+		URL: A string representing the website's main URL for instance "trackor.onevizion.com".
+		userName: the username used to login to the system or OneVizion API Security Token Access Key
+		password: the password used to gain access to the system or OneVizion API Security Token Secret Key
+
+    Exception can be thrown for methods:
+        getNotifQueue,
+        updateNotifQueueRecStatus,
+        addNewAttempt
+    """
+
+    def __init__(self, serviceId, URL="", userName="", password="", paramToken=None, ovToken=False):
+        self._serviceId = serviceId
+        self._URL = URL
+        self._userName = userName
+        self._password = password
+        self._headers = {'content-type': 'application/json'}
+
+        if paramToken is not None:
+            if self._URL == "":
+                self._URL = Config["ParameterData"][paramToken]['url']
+            if self._userName == "":
+                self._userName = Config["ParameterData"][paramToken]['UserName']
+            if self._password == "":
+                self._password = Config["ParameterData"][paramToken]['Password']
+
+        if ovToken == True:
+            self._auth = HTTPBearerAuth(self._userName, self._password)
+        else:
+            self._auth = requests.auth.HTTPBasicAuth(self._userName, self._password)
+
+
+    def getNotifQueue(self):
+        URL = "https://{URL}/api/internal/notif/queue?service_id={ServiceID}".format(URL=self._URL, ServiceID=self._serviceId)
+        OVCall = curl('GET', URL, headers=self._headers, auth=self._auth)
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+        return OVCall.jsonData
+
+    def updateNotifQueueRecStatus(self, notifQueueRecId, status):
+        URL = "https://{URL}/api/internal/notif/queue/{notifQueueRecId}/update_status?status={status}".format(URL=self._URL, notifQueueRecId=notifQueueRecId, status=status)
+        OVCall = curl('PATCH', URL, headers=self._headers, auth=self._auth)
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+
+    def addNewAttempt(self, notifQueueRecId, errorMessage):
+        URL = "https://{URL}/api/internal/notif/queue/{notifQueueRecId}/attempts?error_code={errorMessage}".format(URL=self._URL, notifQueueRecId=notifQueueRecId, errorMessage=errorMessage)
+        OVCall = curl('POST', URL, headers=self._headers, auth=self._auth)
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+
+    def updateNotifQueueRecStatusByObject(self, notifQueueRec):
```

Comment:
Rename updateNotifQueueRecStatusByObject method to updateNotifQueueRecStatus

Reply:
Done